src/BuildScriptGenerator/Helpers/ListBlobsHelper.cs

```diff
@@ -2,22 +2,41 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT license.
 // --------------------------------------------------------------------------------------------
+using System;
 using System.Linq;
+using System.Net.Http;
 using System.Xml.Linq;
 using Microsoft.Oryx.BuildScriptGenerator.Common;
 
 namespace Microsoft.Oryx.BuildScriptGenerator
 {
     internal static class ListBlobsHelper
     {
-        public static XDocument GetAllBlobs(string sdkStorageBaseUrl, string platform, System.Net.Http.HttpClient httpClient, string oryxSdkStorageAccountAccessToken)
+        public static XDocument GetAllBlobs(string sdkStorageBaseUrl, string platform, HttpClient httpClient, string oryxSdkStorageAccountAccessToken)
         {
             var oryxSdkStorageAccountAccessArgs = oryxSdkStorageAccountAccessToken?.TrimStart(new char[] { '?' }) ?? string.Empty;
 
             var url = string.Format(SdkStorageConstants.ContainerMetadataUrlFormat, sdkStorageBaseUrl, platform, string.Empty, oryxSdkStorageAccountAccessArgs);
-            var blobList = httpClient
-                .GetStringAsync(url)
-                .Result;
+            string blobList;
+            try
+            {
+                blobList = httpClient
+                    .GetStringAsync(url)
+                    .Result;
+            }
+
+            // AggregateException is thrown by .Result
+            catch (AggregateException ae)
+            {
+                throw new AggregateException(
+                        $"Http request to '{sdkStorageBaseUrl}' failed. {Constants.NetworkConfigurationHelpText}\n{ae}");
+            }
+
+            if (string.IsNullOrEmpty(blobList))
```

Comment:
I'm just curious: did this happen? Did we sometimes get some empty results from the call? I'm just curious to know if this is a preventive check to make sure this never happens, or if we actually did see that happening sometimes.

Reply:
No, its preventative. Mainly because I need to initialize the `blobList` variable before the try/catch block, and I initialize it to empty. So I just wanted to cover all my bases and basically to a null check before assuming that the http request returned as expected and trying to parse it.

Reply:
OK fair enough, thanks!